packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -149,31 +149,37 @@ typedef void GlobalKeyRemoveListener(GlobalKey key);
 /// Global keys are relatively expensive. If you don't need any of the features
 /// listed above, consider using a [Key], [ValueKey], [ObjectKey], or
 /// [UniqueKey] instead.
+///
+/// You cannot simultaneously include two widgets in the tree with the same
+/// global key. Attempting to do so will assert at runtime.
 @optionalTypeArgs
 abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
-  /// Creates a [LabeledGlobalKey], which is a [GlobalKey] with a label used for debugging.
+  /// Creates a [LabeledGlobalKey], which is a [GlobalKey] with a label used for
+  /// debugging.
   ///
   /// The label is purely for debugging and not used for comparing the identity
   /// of the key.
-  factory GlobalKey({ String debugLabel }) = LabeledGlobalKey<T>._; // the label is purely for debugging purposes and is otherwise ignored
+  factory GlobalKey({ String debugLabel }) = LabeledGlobalKey<T>._;
 
   /// Creates a global key without a label.
   ///
-  /// Used by subclasss because the factory constructor shadows the implicit
+  /// Used by subclasses because the factory constructor shadows the implicit
   /// constructor.
   const GlobalKey.constructor() : super._();
 
   static final Map<GlobalKey, Element> _registry = new Map<GlobalKey, Element>();
-  static final Map<GlobalKey, int> _debugDuplicates = new Map<GlobalKey, int>();
   static final Map<GlobalKey, Set<GlobalKeyRemoveListener>> _removeListeners = new Map<GlobalKey, Set<GlobalKeyRemoveListener>>();
   static final Set<GlobalKey> _removedKeys = new HashSet<GlobalKey>();
+  static final Set<Element> _debugIllFatedElements = new HashSet<Element>();
+  static final Map<GlobalKey, Element> _debugReservations = new Map<GlobalKey, Element>();
 
   void _register(Element element) {
     assert(() {
       if (_registry.containsKey(this)) {
-        final int oldCount = _debugDuplicates.putIfAbsent(this, () => 1);
-        assert(oldCount >= 1);
-        _debugDuplicates[this] = oldCount + 1;
+        assert(element.widget != null);
+        assert(_registry[this].widget != null);
+        assert(element.widget.runtimeType != _registry[this].widget.runtimeType);
+        _debugIllFatedElements.add(_registry[this]);
```

Comment:
Illfated is one word.  If there was something smaller than a nit, this would be it.